core/node/api_server/src/tx_sender/whitelist.rs

```diff
@@ -0,0 +1,188 @@
+                    "Blocking contract deployment in tx {:?}: deployer_address {:?} not whitelisted",
+                    tx.hash(),
+                    deployer_address
+                );
+                return Err(SubmitTxError::DeployerNotInAllowList(deployer_address));
+            }
+        }
+
+        // If all deployment events pass the allowlist check, forward the submission.
+        self.master_pool_sink
+            .submit_tx(tx, execution_outputs, validation_traces)
+            .await
+    }
+}
+
+#[derive(Debug, Deserialize)]
+struct WhitelistResponse {
+    addresses: Vec<Address>,
+}
+
+#[derive(Debug, Clone, Default)]
+pub struct SharedAllowList {
+    inner: Arc<RwLock<HashSet<Address>>>,
+}
+
+impl SharedAllowList {
+    pub fn writer(&self) -> Arc<RwLock<HashSet<Address>>> {
+        Arc::clone(&self.inner)
+    }
+
+    pub async fn is_address_allowed(&self, address: &Address) -> bool {
+        self.inner.read().await.contains(address)
+    }
+}
+
+/// Task that periodically fetches and updates the allowlist from a remote HTTP source.
+#[derive(Debug, Clone)]
+pub struct AllowListTask {
+    url: String,
+    refresh_interval: Duration,
+    allowlist: SharedAllowList,
+    client: Client,
+}
+
+impl AllowListTask {
+    const REQUEST_TIMEOUT: Duration = Duration::from_secs(10);
+    pub fn from_config(deployment_allowlist: DeploymentAllowlist) -> Self {
+        Self {
+            url: deployment_allowlist
+                .http_file_url()
+                .expect("DeploymentAllowlist must contain a URL")
+                .to_string(),
+            refresh_interval: deployment_allowlist.refresh_interval(),
+            allowlist: SharedAllowList::default(),
+            client: Client::new(),
+        }
+    }
+
+    pub fn refresh_interval(&self) -> Duration {
+        self.refresh_interval
+    }
+
+    pub fn shared(&self) -> SharedAllowList {
+        self.allowlist.clone()
+    }
+
+    pub async fn fetch(
+        &self,
+        current_etag: Option<&str>,
+    ) -> anyhow::Result<Option<(HashSet<Address>, Option<String>)>> {
+        let mut request = self.client.get(&self.url).timeout(Self::REQUEST_TIMEOUT);
+
+        if let Some(etag) = current_etag {
+            request = request.header("If-None-Match", etag);
+        }
+
+        let response = request.send().await?;
+
+        if response.status() == reqwest::StatusCode::NOT_MODIFIED {
+            tracing::debug!("Allowlist unchanged (304 Not Modified)");
+            return Ok(None);
+        }
+
+        let response = response.error_for_status()?;
+
+        let new_etag = match response.headers().get("ETag") {
+            Some(value) => match value.to_str() {
+                Ok(s) => Some(s.to_string()),
+                Err(_) => None,
+            },
+            None => None,
+        };
+        let list = response.json::<WhitelistResponse>().await?;
+        let addresses: HashSet<H160> = list.addresses.into_iter().collect();
```

Comment:
Nit: `H160` and `Address` are the same type; there's no need to use both (we mostly use `Address` in the codebase). Here specifically, it's possible to use a wildcard: `HashSet<_>`, since the compiler can infer the type automatically.